staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/mutating/dispatcher.go

```diff
@@ -158,6 +159,18 @@ func (a *mutatingDispatcher) callAttrMutatingHook(ctx context.Context, h *v1beta
 		return apierrors.NewInternalError(err)
 	}
 
+	// If object is mutated, record the mutation in audit log
+	if !bytes.Equal(objJS, patchedJS) {
+		// use a key prefix different from the webhook name to avoid key conflict, where:
+		//     "webhooktrace" is the name of the virtual plugin,
+		//     "admission.k8s.io" is the name of the org,
+		//     webhook name (h.Name) is the key name,
+		//     "mutation" is the value
+		if err := attr.AddAnnotation("webhooktrace.admission.k8s.io/"+h.Name, "mutation"); err != nil {
```

Comment:
Do we want/need the same for rejection of objects in the validation phase (and also here) ?

Reply:
by validation phase do you mean the `if !response.Response.Allowed` block? 

I think other code paths that return errors should already have error message clear enough to indicate the problem is related to webhook.